tasks/task_manager.hh

```diff
@@ -126,6 +128,16 @@ public:
             }
 
             virtual future<> abort() noexcept {
+                if (!_as.abort_requested()) {
+                    _as.request_abort();
+                    return with_semaphore(_children_access_semaphore, 1, [this] () mutable {
+                        return do_for_each(_children, [] (auto& child) {
+                            return smp::submit_to(child.get_owner_shard(), [&child] () {
+                                return child->abort();
+                            });
+                        });
+                    });
+                }
                 return make_ready_future<>();
```

Comment:
This could be simpler written as a coroutine.

Reply:
Note that instead of `with_semaphore` you can use get_units to hold the semaphore while iterating over `_children`.